storage/interceptors/broker_notifications_interceptor.go

```diff
@@ -164,19 +158,79 @@ func fetchBrokerPlans(ctx context.Context, brokerID string, repository storage.R
 	return objList.(*types.ServicePlans).ServicePlans, nil
 }
 
-func getSupportedPlatformsForPlans(plans []*types.ServicePlan) []string {
-	platformTypes := make(map[string]bool)
+func ResolveSupportedPlatformIDsForPlans(ctx context.Context, plans []*types.ServicePlan, repository storage.Repository) ([]string, error) {
```

Comment:
It's weird for other packages to have to import the `interceptors` package to use this function. Did you try moving this somewhere in the `pkg` package?

Reply:
moved to storage/service_plans